skbio/sequence/_sequence.py

```diff
@@ -503,43 +519,8 @@ def __init__(self, sequence, metadata=None,
                 positional_metadata = sequence.positional_metadata
             sequence = sequence._bytes
```

Comment:
Thanks! I think the previous factoring ended up setting `_owns_bytes` from within `_set_sequence` but now it's up to each branch in the constructor to determine it (unless passing through `_set_bytes_contiguous`). This flag lets us know whether a copy of the underlying sequence bytes needs to be made when using the `_byte_ownership` context manager to perform in-place modifications.